### DISCUSSION TITLE
REST API POST Binding Problem

### DISCUSSION THREAD

**Discussion by hci-ikahera** [Budibase/budibase#10946] [Help] [answered]
Hello everyone,
Anybody here can help me about my problem related to REST API POST Binding Problem?
Long story short, I use POST REST API Datasource which is using Binding for the Body. **The Binding value is allowed to be null**.
The strange think I found was if I leave the default value of the Binding empty in the REST API Datasource as the screenshot below,

<img width="1010" alt="Screenshot 2023-06-19 at 15 52 13" src="https://github.com/Budibase/budibase/assets/95213000/21bf97f3-9805-4558-bc71-0b7ba74b4901">

with the Body,

<img width="996" alt="Screenshot 2023-06-19 at 16 09 49" src="https://github.com/Budibase/budibase/assets/95213000/eaf5db81-37fd-4ba6-afab-1d35d375c272">

When I run my App, 
- It works as what I want when I leave the value in the Component -> Execute Query **empty**. However, when **I set a value** in the Component -> Execute Query, instead of using value which I set, **the value always send an empty string for Payload**. 
- If I set the default value of the Binding as **null** in the REST API Datasource, **it send a correct Payload if I set a value in the Component -> Execute Query**. If I leave the value in the Component -> Execute Query **empty**, it always send **null as String for Payload** instead of null as null, and that is not what I want.

Anybody here can give me a solution about this case?
What I want are,
- It sends null as null for Payload when I leave the value in the Component -> Execute Query **empty**
- It sends exactly the value I set in the Component -> Execute Query for Payload if I don't leave it empty

I appreciate any kind of your help.
Thank you.

Regards,
Ika

**Comment by ConorWebb96** [comment]
Hey @hci-ikahera,

Have you binded the fields you are using to post to the rest api on the execute query action? I've supplied an example of this to help you better understand below;
<img width="860" alt="Screenshot 2023-06-19 at 11 16 05" src="https://github.com/Budibase/budibase/assets/126772285/a447d56f-5e97-4cb7-b59c-f599db99ad97">

If not could you supply a screenshot of how your execute query is setup?

I hope this helps.

**Reply by avivkri** [reply, marked as answer]
@hci-ikahera and @ConorWebb96,
I've faced this problem sometimes, what worked for me before was to remove the "Binding." prefix from the binding references.
Example: {{ idNotification }} instead of {{ Binding.idNotification }}

**Reply by hci-ikahera** [reply]
Yes, I have binded the fields in the Execute Query @ConorWebb96 
Oh okay, I will try it @avivkri. Thank you 🙏🏼
